pkg/route/validation/validation.go

```diff
@@ -342,6 +385,53 @@ func validateWildcardPolicy(host string, policy routev1.WildcardPolicyType, fldP
 	return nil
 }
 
+var (
+	notAllowedHTTPHeaders   = []string{"strict-transport-security", "proxy", "cookie", "set-cookie"}
+	notAllowedHTTPHeaderSet = sets.NewString(notAllowedHTTPHeaders...)
+)
+
+func validateHeaders(path *field.Path, headerValues []routev1.RouteHTTPHeader) *field.Error {
+	unique := make(map[string]bool, len(headerValues))
+	us := make([]string, len(unique))
+	for key, value := range headerValues {
+		if len(value.Name) == 0 {
+			return field.Invalid(path.Index(key).Child("name"), value.Name, fmt.Sprintf("name must be present"))
+		}
+
+		if !unique[value.Name] {
+			us = append(us, value.Name)
+			unique[value.Name] = true
+		} else {
+			return field.Invalid(path.Index(key).Child("name"), value.Name, fmt.Sprintf("duplicate header name not allowed"))
+		}
+
+		if value.Action.Set != nil && len(value.Action.Set.Value) > 16384 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Action.Set.Value, fmt.Sprintf("value exceeds the maximum length which is 16384"))
+		}
+		if len(value.Name) > 1024 {
+			return field.Invalid(path.Index(key).Child("name"), value.Name, fmt.Sprintf("name exceeds the maximum length which is 1024"))
+		}
+		if value.Action.Set != nil && len(value.Action.Set.Value) == 0 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Action.Set.Value, fmt.Sprintf("value must be present"))
```

Comment:
Include the name of the name of the header that is missing a value.
```suggestion
			return field.Invalid(path.Index(key).Child("action.set.value"), value.Action.Set.Value, fmt.Sprintf("value must be present for header %s", value.Name))
```

Reply:
Or:
```suggestion
			return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, fmt.Sprintf("header value is missing"))
```

Reply:
This will throw a required error as per Miciah's commit